Packages/vcs/Lib/colorpicker.py

```diff
@@ -0,0 +1,212 @@
+        self.selectedActor.GetProperty().SetLineWidth(3);
+        self.color = color
+        # Make sure the current color is selected
+        self.selectCell(color)
+        self.color_renderer.AddActor(self.selectedActor)
+        self.click_handler = inter.AddObserver("LeftButtonPressEvent", self.leftButtonPressEvent)
+
+        inter.Start()
+
+    def make_current(self):
+        self.render_window.MakeCurrent()
+
+    def topRendererAtPoint(self, x, y):
+        inter = self.render_window.GetInteractor()
+
+        top = None
+
+        for renderer in collection(inter.GetRenderWindow().GetRenderers()):
+            if renderer.IsInViewport(x, y):
+                if top is None or top.GetLayer() <= renderer.GetLayer():
+                    top = renderer
+        return top
+
+    def change_map(self, state):
+        self.colormap = self.colormaps[state]
+        self.colors =  [[int(c / 100.0 * 255.0) for c in self.colormap.index[i]] for i in range(len(self.colormap.index))]
+        colorData = colors_to_scalars(self.colors)
+        self.actor.GetMapper().GetInput().GetCellData().SetScalars(colorData)
+        self.actor.GetMapper().Update()
+        self.render_window.Render()
+
+    def save(self, state):
+        if self.on_save is not None:
+            self.on_save(self.colormap, self.color)
+
+        self.close()
+
+    def cancel(self, state):
+        if self.on_cancel is not None:
+            self.on_cancel()
+
+        self.close()
+
+    def selectCell(self, cellId):
+        if cellId == -1:
+            return
+        ids = vtk.vtkIdTypeArray();
+        ids.SetNumberOfComponents(1);
+        ids.InsertNextValue(cellId);
+
+        selectionNode = vtk.vtkSelectionNode();
+        selectionNode.SetFieldType(vtk.vtkSelectionNode.CELL);
+        selectionNode.SetContentType(vtk.vtkSelectionNode.INDICES);
+        selectionNode.SetSelectionList(ids);
+
+        selection = vtk.vtkSelection();
+        selection.AddNode(selectionNode);
+
+        extractSelection = vtk.vtkExtractSelection();
+
+        extractSelection.SetInputData(0, self.actor.GetMapper().GetInput());
+        extractSelection.SetInputData(1, selection);
+
+        extractSelection.Update();
+
+        selected = vtk.vtkUnstructuredGrid();
+        selected.ShallowCopy(extractSelection.GetOutput());
+
+        self.selectedMapper.SetInputData(selected);
+        self.selectedMapper.Update()
+
+    def leftButtonPressEvent(self, obj, event):
+        inter = self.render_window.GetInteractor()
+
+        x, y = inter.GetEventPosition()
+
+        renderer = self.topRendererAtPoint(x, y)
+
+        if renderer:
+            picker = vtk.vtkCellPicker()
+            picker.SetTolerance(.0005)
+            picker.Pick(x, y, 0, renderer)
+
+            cell = picker.GetCellId()
+            self.selectCell(cell)
+            if cell >= 0:
+                self.color = int(cell)
+
+            self.render_window.Render()
+
+
+    def close(self):
```

Comment:
Hm... that looks kind of like one of the vtkButtonWidgets is still hooked up to receive events from the interactor that's being dismantled. Which seems weird, because when I call self.save_button.detach() and self.cancel_button.detach(), it should wind up calling widget.py's detach methods, which look like this:

```python
    def detach(self):
        render = self.widget.GetCurrentRenderer()
        if render is None:
            return

        if render.HasViewProp(self.repr):
            render.RemoveViewProp(self.repr)
        self.unsubscribe(*self.subscriptions.keys())
        self.widget.SetCurrentRenderer(None)
        self.widget.SetInteractor(None)
        self.widget.Off()
        get_manager(self.interactor).remove_widget(self)
```
`unsubscribe()` will unsubscribe the widget from all events that it's subscribed to

`get_manager()` will return an instance of InterfaceManager, (located in manager.py) and remove the reference to the button object

Is there a more appropriate way to clean up a vtkWidget / vtkWidgetRepresentation? I kind of had to feel my way through getting this method to work.